### PR TITLE
create_expression_profile(): promote global molecule scalars to plain arguments

### PR DESCRIPTION
**Pull request by Felixmil**
Closes #169

This change promotes the three fixed global molecule-properties scalars that PK-Sim exposes in the expression-profile molecule-properties panel to plain, unit-aware arguments on `create_expression_profile()`, following the promotion pattern already used elsewhere in the package. Previously a user who wanted to set any of them had to hand-author a raw `LocalizedParameter` entry in the `parameters` list, keyed by a `{molecule}|{parameter name}` `Path`. The `parameters` escape hatch stays available and unchanged, and a promoted argument produces the same snapshot JSON as the equivalent hand-authored entry.

## New arguments

`create_expression_profile()` gains three optional numeric scalar arguments, each with a paired unit argument (all inserted after `parameters`, all defaulted so existing calls are unaffected):

- `reference_concentration` (+ `reference_concentration_unit`, default `"µmol/l"`, dimension `"Concentration (molar)"`), mapped to the `Path` leaf `"Reference concentration"`.
- `half_life_liver` (+ `half_life_liver_unit`, default `"h"`, dimension `"Time"`), mapped to the `Path` leaf `"t1/2 (liver)"`.
- `half_life_intestine` (+ `half_life_intestine_unit`, default `"h"`, dimension `"Time"`), mapped to the `Path` leaf `"t1/2 (intestine)"`.

The reference-concentration default unit is the Greek-micro `"µmol/l"` (U+00B5), which is what `ospsuite` returns for the `"Concentration (molar)"` dimension and what the repository fixture stores; the ASCII `"umol/l"` is not a valid unit.

## Behaviour

Each value argument defaults to `NULL`, meaning "not supplied": when all three are omitted the factory emits no `Parameters` key and the output is identical to before, preserving full backward compatibility. When a value is supplied it is validated as a single finite numeric, its paired unit is validated with `validate_unit()` against the parameter's dimension, and a `list(Path = "{molecule}|{leaf}", Value = <value>, Unit = <unit>)` entry is folded into `data$Parameters`. Promoted entries come first (reference concentration, liver half-life, intestine half-life), followed by the entries resolved from `parameters` in their given order. A promoted unit argument supplied without its value argument is a silent no-op. Supplying the same parameter both through a promoted argument and through a `parameters` entry (whether that entry carries `Path` directly or only `Name`, which resolves to the same `Path`) aborts with a clear conflict error naming the argument and the `Path`.

## Files touched

- `R/create_expression_profile.R`: added the six arguments and their roxygen `@param` docs, an `@examples` block using a promoted argument, and a `@description` note; reworked the `parameters` handling to resolve the raw entries, build the promoted entries, merge them (promoted first), and assign `data$Parameters` only when non-empty.
- `R/utils-create.R`: added the internal helper `build_promoted_molecule_parameters()`, which validates value and unit, detects `Path` conflicts against the resolved parameters, and returns the promoted `LocalizedParameter` entries as an unnamed list.
- `tests/testthat/test-create_expression_profile.R` and `tests/testthat/_snaps/create_expression_profile.md`: eleven new self-contained tests (each promoted argument's shape, a non-default unit, invalid unit, invalid value, both conflict variants, the parameters-only no-conflict case, merge order, and no `Parameters` when omitted), plus a byte-identity assertion between a promoted argument and the hand-authored equivalent.
- `man/create_expression_profile.Rd`: regenerated with `devtools::document()`.
- `NEWS.md`: one development-version bullet for the new arguments, placed alphabetically.

## Verification

Run with `NOT_CRAN=true`:

- `tests/testthat/test-create_expression_profile.R`: all tests pass; new error snapshots recorded and reviewed (invalid unit lists valid units, invalid value names the argument, conflict names the argument and the `Path`).
- `devtools::test(filter = "create_expression_profile|expression-profile")`: green.
- Full suite `devtools::test()`: 3124 passing. The only failures (3, all in `test-snapshot.R`'s "rejects snapshots newer than the supported ceiling" case) are a pre-existing version-string drift on the base branch: `DESCRIPTION` is at `1.0.0.9002` while `tests/testthat/_snaps/snapshot.md` still expects `1.0.0.9001`. Both files are unrelated to this issue and untouched here; the drift reproduces on `origin/main` and is out of scope for this change.

# QA: Approved

The implementation promotes the three global molecule scalars onto `create_expression_profile()` exactly as specified: unit-aware arguments with the correct Greek-micro default, byte-identical JSON to the `parameters` escape hatch, deterministic promoted-first ordering, and both conflict variants rejected. All 154 tests in the two named files pass with `NOT_CRAN=true`; the only suite failures are the documented pre-existing version drift the PR does not touch.

<details>
<summary>Full QA report</summary>

## Scope and method

Reviewed `gh pr diff 173` against `spec.md` and `plan.md`. Confirmed the PR's changed files are exactly the six expected (`NEWS.md`, `R/create_expression_profile.R`, `R/utils-create.R`, `man/create_expression_profile.Rd`, `tests/testthat/_snaps/create_expression_profile.md`, `tests/testthat/test-create_expression_profile.R`), so it touches neither `DESCRIPTION` nor `tests/testthat/_snaps/snapshot.md`. Ran the two named test files from the build worktree, and executed a small runtime probe to verify byte-identity, ordering, and the unit-only no-op behaviour directly rather than trusting the diff alone.

## Acceptance criteria mapping

1. Three value args plus paired `_unit` args, defaults `"µmol/l"` / `"h"` / `"h"`. Met. Signature adds all six after `parameters`. The reference-concentration default was verified at the byte level: `grep -o 'µmol/l' | xxd` shows `c2 b5` (U+00B5 Greek micro), not ASCII `umol/l`. This is the spec's single highest-risk defect and it is correct.
2. Correct leaf strings, value, and unit per promoted arg. Met. Runtime probe: `reference_concentration = 4.32` yields `list(Path = "CYP3A4|Reference concentration", Value = 4.32, Unit = "µmol/l")`; `half_life_liver` and `half_life_intestine` map to `t1/2 (liver)` / `t1/2 (intestine)` with `Unit = "h"`. Leaf strings reproduced verbatim including spaces, slash, and parentheses.
3. `Parameters[]` unnamed list, contains promoted plus `parameters` entries. Met. `is.null(names(profile$data$Parameters))` is `TRUE`; the merge test produces length-2 output with both entries.
4. Invalid promoted unit aborts via `validate_unit()`. Met. Snapshot shows `Error in validate_unit()` for both `reference_concentration_unit = "umol/l"` (ASCII) and `half_life_liver_unit = "bogus"`, each listing the valid units for the dimension.
5. Invalid promoted value (NA / non-numeric / non-finite / length != 1) aborts naming the arg. Met. The `is.numeric && length == 1 && is.finite` guard covers all four; snapshots for `NA`, `"x"`, `c(1, 2)`, `Inf` all read ``! `reference_concentration` must be a single finite numeric value``.
6. Same parameter via both a promoted arg and a `parameters` entry aborts, for both the Path and Name-only variants. Met. Conflict detection runs against `existing_paths` derived from `to_raw_parameters(..., "Path")`, which copies `Name` to `Path`; both snapshot cases produce the conflict error naming the arg and the Path and stating the two mutually exclusive ways.
7. Supplying one of the three only via `parameters` is not a conflict and yields unchanged output. Met. Dedicated test asserts the single hand-authored entry is returned unchanged with no error.
8. Omitting all promoted args yields no new `Parameters` entry. Met. The `if (length(combined) > 0)` guard preserves absence; test asserts `expect_null(profile$data$Parameters)`. Runtime probe with a lone `_unit` arg (value `NULL`) also confirms the no-op: no Parameters, no validation triggered.
9. Existing tests pass unchanged; new tests cover the named cases. Met. `test-create_expression_profile.R` runs `[ FAIL 0 | PASS 51 ]`; `test-expression-profile.R` runs `[ FAIL 0 | PASS 103 ]`. New tests cover each promoted arg's shape, a non-default unit (`min`), invalid unit, invalid value, both conflict variants, the parameters-only no-conflict case, merge order, and the no-Parameters-when-omitted case. No existing test was edited.
10. Structured surface unchanged. Met. The diff only replaces the `parameters` handling block; `build_expression_containers`, `build_disease_state`, and the `ExpressionProfile$new(data)` path are untouched. The 103-test expression-profile suite passing confirms no behavioural regression.
11. `man/*.Rd` and `NAMESPACE` regenerated, not hand-edited; roxygen prose wrapped at 80. Met. The Rd is regenerated (matches roxygen source); `NAMESPACE` is unchanged (no new export, `validate_unit()` already exported, so no regeneration delta is expected). No roxygen comment line in the file exceeds 80 columns.
12. NEWS gains one alphabetically placed single-line bullet ending `(#169).`. Met. The bullet sits at the top of the development-version section (sorts before `create_snapshot()` and `fraction_unbound()`), is a single line, present tense, positive, with backticked identifiers and `(#169).` at the end.
13. Tests green with `NOT_CRAN=true`. Met for this PR's scope (see baseline note below).

## Convention checks

- Base pipe only, no `%>%` in the touched R files.
- Errors use `cli::cli_abort()` with `{.arg ...}` style throughout the new helper.
- No dash-as-punctuation and no em-dash in NEWS, roxygen prose, or the helper comment.
- Commit message has no `Co-Authored-By` trailer and follows the per-file bullet structure.
- Exported-before-internal ordering respected: `build_promoted_molecule_parameters()` is an internal helper added inside the all-internal `R/utils-create.R`, placed after `to_raw_parameters()`.

## Adversarial and edge-case findings

- Byte-identity (FR-4) verified at runtime, not just by test: `identical(promoted$data$Parameters, hand$data$Parameters)` is `TRUE`.
- Ordering (FR-5): runtime probe confirms the promoted entry is first and the `parameters` entry second.
- Unit-only-without-value no-op (decision 3): confirmed a bogus `_unit` with `NULL` value neither validates nor emits, so harmless defaults stay harmless.
- Name-only conflict (edge case): correctly caught because conflict detection compares against resolved Paths (`Name` copied to `Path`), not the raw pre-resolution list.
- No unintended argument-order regression: the six args are appended after `parameters`, and no existing test or example passes `expression` / `disease` / `description` positionally.
- No defect found in scope.

## Pre-existing baseline issue (not grounds for rejection)

The full suite has 3 failures in `test-snapshot.R` ("rejects snapshots newer than the supported ceiling"): `DESCRIPTION` is `1.0.0.9002` while `_snaps/snapshot.md` hardcodes `v1.0.0.9001`. Confirmed the PR touches neither file (`git diff --stat` for both is empty), so this drift predates the PR and reproduces on `origin/main`. It is out of scope for issue #169 and is noted here only for completeness.

</details>

## Summary by CodeRabbit

* **New Features**
  * Added optional convenience arguments for setting reference concentration and liver or intestine half-life values directly when creating expression profiles.
  * Added support for custom units for each new value, with sensible defaults.
  * Promoted values are included in generated profile parameters and remain compatible with existing parameter configuration.

* **Bug Fixes**
  * Added validation for invalid values and units.
  * Added clear conflict detection when promoted values duplicate existing parameters.

### CODE REVIEW

File: R/create_expression_profile.R [hunk 1]
```diff
@@ -85,6 +123,12 @@ create_expression_profile <- function(
   transport_type = NULL,
   ontogeny = NULL,
   parameters = NULL,
+  reference_concentration = NULL,
+  reference_concentration_unit = "µmol/l",
+  half_life_liver = NULL,
+  half_life_liver_unit = "h",
+  half_life_intestine = NULL,
+  half_life_intestine_unit = "h",
```
[the remark this answers is not in the document]
Comment (Felixmil) [reply R/create_expression_profile.R]: Fixed in e674b6a. The promoted scalar arguments (reference_concentration and its unit, half_life_liver and its unit, half_life_intestine and its unit) now sit after the pre-existing arguments, so expression, disease, and description keep their original positions. Added a positional-call regression test that binds those slots through the pre-existing signature order.

File: R/create_expression_profile.R [hunk 2]
```diff
@@ -119,11 +163,49 @@ create_expression_profile <- function(
...
+        unit = half_life_intestine_unit,
+        dimension = "Time",
+        leaf = "t1/2 (intestine)"
+      )
+    ),
+    existing_paths = existing_paths
+  )
+  combined <- c(promoted, raw_params)
+  if (length(combined) > 0) {
+    data$Parameters <- combined
```
[the remark this answers is not in the document]
Comment (Felixmil) [reply R/create_expression_profile.R:209]: Fixed in e674b6a. We now capture parameters_supplied <- !is.null(parameters) before normalization and gate the assignment on parameters_supplied || length(combined) > 0. An explicit parameters = list() therefore keeps an empty Parameters array (serialized as []), while omitting parameters entirely (NULL) still omits the key. Added a test covering both cases.
